src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -301,9 +302,22 @@ public void Complete()
                 _buf.EndCopyMode();
 
                 _connector.SendMessage(CopyDoneMessage.Instance);
-                Expect<CommandCompleteMessage>(_connector.ReadMessage());
+
+                var msg1 = _connector.ReadMessage();
```

Comment:
`Expect<>` already returns the message cast to the right type, no need to read it, call `Expect<>` on it and then downcast to `CommandCompleteMessage` below.

Reply:
This is helpful. I'll revert with some improvements soon.  Thanks again!